lib/web/conn_upgrade.go

```diff
@@ -236,35 +237,46 @@ func (conn *waitConn) Close() error {
 }
 
 type websocketALPNServerConn struct {
-	*websocket.Conn
+	net.Conn
 	readBuffer []byte
 	readError  error
 	readMutex  sync.Mutex
 	writeMutex sync.Mutex
-}
 
-func newWebSocketALPNServerConn(wsConn *websocket.Conn) *websocketALPNServerConn {
-	return &websocketALPNServerConn{
-		Conn: wsConn,
-	}
+	logContext context.Context
+	logger     *slog.Logger
 }
 
-func (c *websocketALPNServerConn) convertError(err error) error {
-	if websocket.IsCloseError(err,
-		websocket.CloseAbnormalClosure,
-		websocket.CloseGoingAway,
-		websocket.CloseNormalClosure,
-	) {
-		return io.EOF
+func newWebSocketALPNServerConn(ctx context.Context, wsConn *websocket.Conn, logger *slog.Logger) *websocketALPNServerConn {
```

Comment:
Maybe we can require as a parameter pain `net.Conn` instead of `websocket.Conn`? The only thing we do with it is extracting net Conn, so caller can do that.